### PR TITLE
Make it possible to use YAML data in lessons + 3.7/mac/Windows CI

### PR DESCRIPTION
**Pull request by encukou**
With this, we can keep structured info about tasks in YAML, but render them as a regular lesson.

This means we can use [courseware](https://projekty.pyladies.cz/) [task definitions](https://github.com/messa/pyladies-courseware/blob/master/data/2018_pyladies_praha_autumn/homeworks/handout1.yaml) and stick them in HTML.

Of course, in the future, structured data on tasks should be added to the API.

I'll follow up with a `pyladies/naucse.python.cz` PR for testing.

### CODE REVIEW

**Comment by encukou** [comment]
https://github.com/PyLadiesCZ/naucse.python.cz/pull/21 shows the intended usage.

**Comment by encukou** [comment]
I've added two commits that:
- add Travis CI runs for Python 3.7, macOS and Windows
- make all paths in the output go through PurePosixPath (so they have the `/` path separator), so output is the same on all the platforms.

I put them in the same PR because they'd conflict if I rebased them onto master. But if they're not good, the first two commits can be merged separately.

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
+    - os: linux
+      # Get Python 3.7 from Xenial
+      # https://github.com/travis-ci/travis-ci/issues/9815
+      python: 3.7
+      dist: xenial
+    - os: windows
+      # Get Python 3.7 from Chocolatey
+      language: sh
+      python: 3.7
+      env: PYTHON=py
```
Comment (hroncok) [inline .travis.yml]: I suppose using just `python` on Windows didn't work?

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
+      before_install:
+        - choco install python --version 3.7
+        - export PATH="/c/Python37:/c/Python37/Scripts:$PATH"
+        - python -m pip install --upgrade pip wheel
```
Comment (hroncok) [inline .travis.yml]: ...because here you do use it.

File: .travis.yml [hunk 3]
```diff
@@ -1,18 +1,46 @@
...
 install:
-- python -m pip install . 'pytest>=4.0'
+- $PYTHON -m pip install . 'pytest>=4.0'
 
 script:
-- python -m pytest test_naucse_render
-
-after_script:
-- echo "Suplementary logs:"
-- cat .arca/naucse.log
```
Comment (hroncok) [inline .travis.yml]: This is no longer useful?

File: naucse_render/encode.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    elif isinstance(value, dict):
+        # Dicts: Convert keys and values; also ensure keys are str
+        return {
+            str(encode_for_json(k)): encode_for_json(v)
+            for k, v in value.items()
+        }
+
+    elif isinstance(value, (list, tuple)):
+        # Sequences: Convert all items
```
Comment (hroncok) [inline naucse_render/encode.py]:
```suggestion
        # Sequences: Convert all items and always use lists
```

File: naucse_render/encode.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    elif isinstance(value, str):
+        # Convert all str subclasses (like Markup) to actual str
+        return str(value)
+
+    elif isinstance(value, (int, bool, type(None))):
+        return value
+
+    # Note: Floats are inexact, and avoided intentionally. Add them if needed.
+    raise TypeError(value)
```
Comment (hroncok) [inline naucse_render/encode.py]:
```suggestion
    raise TypeError(
        f'{value} ({type(value).__name__}) not supported by encode_for_json()')
```

File: .travis.yml (hunk 1 above)
Comment (encukou) [reply .travis.yml]: It works! Thanks :)
As you can imagine, getting this to work was a lot of trial and error. After that I tend to be blind to the details :)

File: .travis.yml (hunk 3 above)
Comment (encukou) [reply .travis.yml]: Thanks for catching that.

Comment (encukou) [reply .travis.yml]: Actually, it is not useful here – `naucse_render` is often called from within Arca, but it (and its tests) doesn't use Arca.

**Comment by encukou** [comment]
Thanks for the review!
If you have time for another check it would be great, otherwise I'll merge today.

**Pull request merged**